openerp/tools/convert.py

```diff
@@ -702,6 +702,8 @@ def _tag_record(self, rec, data_node=None, mode=None):
                         f_val = int(f_val)
                     elif model._fields[f_name].type in ['float', 'monetary']:
                         f_val = float(f_val)
+                    elif model._fields[f_name].type == 'boolean' and isinstance(f_val, basestring):
+                        f_val = f_val.lower() in ['1', 'true', 'yes']
```

Comment:
Sounds good, thanks!
If you don't mind, I'd rather have it consistent with the current behavior of [`nodeattr2bool()`](https://github.com/vauxoo-dev/odoo/blob/0691d8bf51ca6efe170946696456015c7bbf9856/openerp/tools/convert.py#L208) though, as we use it for other boolean XML attributes. It will also be slightly more backwards-compatible, as only values matching a "falsy" string will suddenly be evaluated differently.